File: Data Structures/Stack&Queues/queues.py
# Queues - FIFO

from collections import deque

# Initialize deque (double-ended queue)
d = deque()  # O(1), initializing an empty deque

# Adding elements to the queue
d.append(1)  # O(1), appending to the end
d.append(2)  # O(1)
d.append(3)  # O(1)

print(d)  # O(n), printing the entire queue (where n is the number of elements in the queue)

# Pop from left as standard queue (dequeue operation)
d.popleft()  # O(1), removing from the front

print(d)  # O(n), printing the queue after pop

# Display the front element of the queue (peek operation)
print(d[0])  # O(1), accessing the first element

# Display the last element of the queue (rear view)
print(d[-1])  # O(1), accessing the last element

File: Data Structures/Stack&Queues/queues.cpp
#include <iostream>
#include <queue>

int main(){
std::queue<int> q;

//push element in queue
q.push(1);
q.push(2);
q.push(3);

//pop element from queue
q.pop();

//view front and last element of queue
std::cout << "front element: " << q.front() << "\n";
std::cout << "back element: " << q.back() << "\n";
return 0;
}
